apps/src/templates/courseOverview/ResourcesEditor.js

```diff
@@ -57,11 +78,14 @@ export default class ResourcesEditor extends Component {
   }
 
   render() {
-    const { resources } = this.state;
+    const { resources, errorString } = this.state;
+
+    // avoid showing multiple empty reosurces
+    const lastNonEmpty = _.findLastIndex(resources, ({type, link}) => link && type);
 
     return (
       <div>
-        {resources.map((resource, index) =>
+        {resources.slice(0, lastNonEmpty + 2).map((resource, index) =>
```

Comment:
This is mostly a function of the steps I took in iterating to get to this point. If I was starting from scratch, it would probably be cleaner to have resources match the DOM elements, and then look at maxResources when we try to add a new one. Not sure it's worth going back and redoing this all at this point though.